core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -256,7 +256,7 @@ static void addFilePermissions(Permissions policy, Environment environment) {
         for (Path path : environment.dataFiles()) {
             addPath(policy, Environment.PATH_DATA_SETTING.getKey(), path, "read,readlink,write,delete");
         }
-        // TODO: this should be removed in ES 6.0!
+        // TODO: this should be removed in ES 6.0! We will no longer support data paths with the cluster as a folder
```

Comment:
I usually try to write a test that fails if the version is 6.0 to ensure we don't miss this, like a simple unittest